### ISSUE TITLE
NullPointerException at startup with screen locked

### ISSUE THREAD

**Issue by adrcotfas** [BeksOmega/looping-layout#39]
* **Device**: Samsung SM-A320FL
* **OS Version**: 8.0.0
* **Project Version**: 0.3.1

### :bug: Describe
A fatal exception occurs when I start an app using this library and the phone is locked with the screen off. It does not occur when the screen is on.

```
kotlin.KotlinNullPointerException
        at com.bekawestberg.loopinglayout.library.LoopingLayoutManager.getInitialItem(LoopingLayoutManager.kt:380)
        at com.bekawestberg.loopinglayout.library.LoopingLayoutManager.onSaveInstanceState(LoopingLayoutManager.kt:163)
        at androidx.recyclerview.widget.RecyclerView.onSaveInstanceState(RecyclerView.java:1384)
        at android.view.View.dispatchSaveInstanceState(View.java:18792)
        at android.view.ViewGroup.dispatchFreezeSelfOnly(ViewGroup.java:3925)
        at androidx.recyclerview.widget.RecyclerView.dispatchSaveInstanceState(RecyclerView.java:1411)
        at android.view.ViewGroup.dispatchSaveInstanceState(ViewGroup.java:3911)
        at android.view.ViewGroup.dispatchSaveInstanceState(ViewGroup.java:3911)
        at android.view.ViewGroup.dispatchSaveInstanceState(ViewGroup.java:3911)
        at android.view.View.saveHierarchyState(View.java:18775)
        at androidx.fragment.app.FragmentStateManager.saveViewState(FragmentStateManager.java:466)
        at androidx.fragment.app.FragmentStateManager.saveBasicState(FragmentStateManager.java:441)
        at androidx.fragment.app.FragmentStateManager.saveState(FragmentStateManager.java:400)
        at androidx.fragment.app.FragmentStore.saveActiveFragments(FragmentStore.java:143)
        at androidx.fragment.app.FragmentManager.saveAllState(FragmentManager.java:2386)
        at androidx.fragment.app.Fragment.performSaveInstanceState(Fragment.java:2865)
        at androidx.fragment.app.FragmentStateManager.saveBasicState(FragmentStateManager.java:434)
        at androidx.fragment.app.FragmentStateManager.saveState(FragmentStateManager.java:400)
        at androidx.fragment.app.FragmentStore.saveActiveFragments(FragmentStore.java:143)
        at androidx.fragment.app.FragmentManager.saveAllState(FragmentManager.java:2386)
        at androidx.fragment.app.FragmentController.saveAllState(FragmentController.java:152)
        at androidx.fragment.app.FragmentActivity.onSaveInstanceState(FragmentActivity.java:512)
        at androidx.appcompat.app.AppCompatActivity.onSaveInstanceState(AppCompatActivity.java:544)
        at android.app.Activity.performSaveInstanceState(Activity.java:1555)
        at android.app.Instrumentation.callActivityOnSaveInstanceState(Instrumentation.java:1392)
        at android.app.ActivityThread.callCallActivityOnSaveInstanceState(ActivityThread.java:4943)
        at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:4231)
        at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4290)
        at android.app.ActivityThread.-wrap25(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1730)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6944)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```
The way I'm using it:
```kotlin
class NumberPicker(
    context: Context,
    private val recyclerView: RecyclerView,
    data: ArrayList<Int>,
    ...
) : NumberPickerAdapter.Listener {

    private val viewManager : LoopingLayoutManager = LoopingLayoutManager(context)
    private val viewAdapter : NumberPickerAdapter =
        NumberPickerAdapter(context, this, prefixWithZero, textSize, textColor)
    private val snapHelper : LoopingSnapHelper = LoopingSnapHelper()

    init {
        viewAdapter.data = data
        snapHelper.attachToRecyclerView(recyclerView)
        recyclerView.apply {
            layoutManager = viewManager
            adapter = viewAdapter
        }

        scrollToPosition(data.indexOf(default))
        ...
    }
```

### :snail: Steps to reproduce

1. Connect the phone through USB for debugging and lock the screen
2. Run the app using this library from Android Studio
3. Check the logs

### :butterfly: Expected behavior

No crash occurs.

### :bee: Requested assignment

I'm just reporting this problem. I don't know what could be the cause here.

**Comment by BeksOmega** [comment]
Thanks for the report @adrcotfas :D Just so you know, it may be a bit before I have time to look into this. But it is now on my to-do list!

**Comment by BeksOmega** [comment]
Had more time today than I thought I did hehe.

Should be fixed in 0.3.2, but jsyk it might take a bit for maven central to get the new version =)

Hope that helps! And thanks again for reporting!
